pkg/cmd/cli/cmd/set/env.go

```diff
@@ -119,6 +128,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	//overwrite := kcmdutil.GetFlagBool(cmd, "overwrite")
 	resourceVersion := kcmdutil.GetFlagString(cmd, "resource-version")
 	outputFormat := kcmdutil.GetFlagString(cmd, "output")
+	secretName := kcmdutil.GetFlagString(cmd, "from-secret")
```

Comment:
We haven't discussed it, but maybe something like `--keys=<comma separate list of keys>`?